bravado/testing/integration_test.py

```diff
@@ -245,7 +245,7 @@ def wait_unit_service_starts(url, max_wait_time=10):
     @pytest.fixture(scope='session')
     def not_answering_http_server(self):
         # Nobody could listen on such address, so TCP 3 way handshake will fail
-        yield 'http://0.0.0.0'
+        yield 'http://0.0.0.0:8080'
```

Comment:
This is still tricky, as that port could be in use as well. We typically use the ephemeral-port-reserve package for such cases.

Is this case really worth doing an "integration test" for? Especially since in the end you're using mock to set the side effect anyway; seems like doing a more unit-testy approach would work better (i.e. no actual TCP connection ever attempted).

Reply:
Also, just for reference, listening on 0.0.0.0 typically means "listen on all interfaces". So you can very much listen on that address, you just can't use it as a target when connecting.

Reply:
There are two tests that are using that fixture
 * `test_connection_errors_are_thrown_as_BravadoConnectionError` which sends a real http request (and TCP connection fails)
 * `test_connection_errors_are_catchable_with_original_exception_types` which sends a request, but the the future is mocked out so it does not really matter

About the IP address, you're right that `0.0.0.0` means _listen on all interfaces_  and apparently this gets routed to `127.0.0.1`.
```
maci@dev31-uswest1adevc:~$ curl -v 0.0.0.0
* Rebuilt URL to: 0.0.0.0/
* Hostname was NOT found in DNS cache
*   Trying 0.0.0.0...
* Connected to 0.0.0.0 (127.0.0.1) port 80 (#0)
> GET / HTTP/1.1
> User-Agent: curl/7.35.0
> Host: 0.0.0.0
> Accept: */*
```

The main issue in using `ephemeral-port-reserve` is that in order to reserve a port the package accepts requests on it and this makes complicated to have a refused TCP connection in a reliable way.

I'm going to change the way this fixture is implemented to ensure that a "test handled" port is used so we don't get influenced by the running environment